### PR TITLE
feat: github integration with Github app auth

### PR DESCRIPTION
**Pull request by sheensantoscapadngan**
# Description 📣
This PR will allow users to setup integration with Github via app authorization. This will allow the creation of a native integration where the access scope is limited only to secret-related permissions. With app authorization, the github integration won't be linked to a specific github user, unlike Oauth.

image
image

## Type ✨

- [ ] Bug fix
- [x] New feature
- [ ] Improvement
- [ ] Breaking change
- [ ] Documentation

# Tests 🛠️

```sh
# Here's some code block to paste some code snippets
```

---

- [x] I have read the [contributing guide](https://infisical.com/docs/contributing/getting-started/overview), agreed and acknowledged the [code of conduct](https://infisical.com/docs/contributing/getting-started/code-of-conduct). 📝

### CODE REVIEW

File: docs/integrations/cicd/githubactions.mdx
Comment (scott-ray-wilson) [inline docs/integrations/cicd/githubactions.mdx:1]: nit: references to `Github` that should be `GitHub`

File: frontend/src/pages/integrations/github/auth-mode-selection.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+export default function GithubIntegrationAuthModeSelectionPage() {
+  const router = useRouter();
+  const { data: cloudIntegrations } = useGetCloudIntegrations();
+  const githubIntegration = cloudIntegrations?.find((integration) => integration.slug === "github");
+  const [selectedAuthMethod, setSelectedAuthMethod] = useState<AuthMethod>(AuthMethod.APP);
+
+  return (
+    <div className="flex h-full w-full items-center justify-center">
+      <Head>
+        <title>Select Github Integration Auth</title>
```
Comment (scott-ray-wilson) [inline frontend/src/pages/integrations/github/auth-mode-selection.tsx:36]: nit: should be `GitHub`

File: frontend/src/pages/integrations/github/auth-mode-selection.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+          <div className="flex flex-row items-center">
+            <div className="flex items-center pb-0.5">
+              <Image
+                src="/images/integrations/GitHub.png"
+                height={30}
+                width={30}
+                alt="Github logo"
+              />
+            </div>
+            <span className="ml-2.5">Github Integration </span>
```
Comment (scott-ray-wilson) [inline frontend/src/pages/integrations/github/auth-mode-selection.tsx:53]: nit: should be `GitHub`

File: frontend/src/pages/integrations/github/auth-mode-selection.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        <CardBody>
+          <FormControl label="Select authentication method">
+            <Select
+              value={selectedAuthMethod}
+              onValueChange={(val) => {
+                setSelectedAuthMethod(val as AuthMethod);
+              }}
+              className="w-full border border-mineshaft-500"
+            >
+              <SelectItem value={AuthMethod.APP}>Github App (Recommended)</SelectItem>
```
Comment (scott-ray-wilson) [inline frontend/src/pages/integrations/github/auth-mode-selection.tsx:77]: nit: should be `GitHub`

File: frontend/src/pages/integrations/github/auth-mode-selection.tsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                  window.location.assign(
+                    `https://github.com/login/oauth/authorize?client_id=${githubIntegration?.clientId}&response_type=code&scope=repo,admin:org&redirect_uri=${window.location.origin}/integrations/github/oauth2/callback&state=${state}`
+                  );
+                }
+              }}
+              colorSchema="primary"
+              variant="outline_bg"
+              className="mt-4 ml-auto w-min"
+            >
+              Connect to Github
```
Comment (scott-ray-wilson) [inline frontend/src/pages/integrations/github/auth-mode-selection.tsx:99]: nit: should be `GitHub`

**Review by scott-ray-wilson — CHANGES_REQUESTED** [review]
Looking good other than a few things pointed out below. Using previous OAuth version of GitHub integration also worked as expected

File: backend/src/services/integration-auth/integration-sync-secret.ts
[the remark this answers is not in the document]
Comment (sheensantoscapadngan) [reply backend/src/services/integration-auth/integration-sync-secret.ts:1]: oh nice catch! I've updated the self-hosted docs to include the missing repository permission: Environments

image

File: frontend/src/pages/integrations/github/auth-mode-selection.tsx
[the remark this answers is not in the document]
Comment (sheensantoscapadngan) [reply frontend/src/pages/integrations/github/auth-mode-selection.tsx:1]: To test it with Github app, you will need to use a connection where the Github app is installed to an organization

File: backend/src/services/integration/integration-service.ts [hunk 5]
```diff
@@ -258,27 +258,7 @@ export const integrationServiceFactory = ({
...
-      if (integrations.length === 0) {
-        // no other integration shares the same integration auth
-        // -> delete the integration auth
-        await integrationAuthDAL.deleteById(integration.integrationAuthId, tx);
-      }
-
-      return deletedIntegrationResult;
-    });
-
+    const deletedIntegration = await integrationDAL.deleteById(id);
```
[the remark this answers is not in the document]
Comment (sheensantoscapadngan) [reply backend/src/services/integration/integration-service.ts:261]: this affects other integrations as well... basically we won't be deleting the integrationAuth anymore unless it's explicitly revoked

File: frontend/src/pages/integrations/select-integration-auth.tsx
[the remark this answers is not in the document]
Comment (sheensantoscapadngan) [reply frontend/src/pages/integrations/select-integration-auth.tsx:1]: I see. I updated the label to be clearer. Does this help? 
image

**Review by scott-ray-wilson — APPROVED** [review]

**Pull request merged**